File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.cs
using System;

public class Solution {
    public int solution(long num) {
        if(num == 1) return 0;
        
        int count = 0;
        
        do{
            num = num % 2 == 0 ? num / 2 : num * 3 + 1;
            count++;
        }while(num != 1);
        
        return count > 500 ? -1 : count;
    }
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int num) 
{   
    long temp  = num;
    int count = 0;
    
    while(temp != 1)
    {
        temp = temp % 2 == 0 ? temp / 2 : temp * 3 + 1;
        count++;
    }
    
    return count > 500 ? -1 : count;
}